pkg/test/v2/register.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2018 Envoyproxy Authors
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+// Package test contains test utilities
+package test
+
+import (
+	"google.golang.org/grpc"
+
+	clusterservice "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	endpointservice "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	listenerservice "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	routeservice "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	accessloggrpc "github.com/envoyproxy/go-control-plane/envoy/service/accesslog/v2"
+	discoverygrpc "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v2"
+	runtimeservice "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v2"
+	secretservice "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v2"
+	"github.com/envoyproxy/go-control-plane/pkg/server/v2"
+)
+
+// RegisterAccessLogServer starts an accessloggrpc service.
+func RegisterAccessLogServer(grpcServer *grpc.Server, als *AccessLogService) {
+	accessloggrpc.RegisterAccessLogServiceServer(grpcServer, als)
+}
+
+// RegisterServer registers with v2 services.
+func RegisterServer(grpcServer *grpc.Server, server server.Server) {
+	// register services
+	discoverygrpc.RegisterAggregatedDiscoveryServiceServer(grpcServer, server)
+	endpointservice.RegisterEndpointDiscoveryServiceServer(grpcServer, server)
```

Comment:
@alecholmez why is ADS and individual xDS registered? The interface passed to `RegisterAggregatedDiscoveryServiceServer` must implement `DeltaAggregatedResources(AggregatedDiscoveryService_DeltaAggregatedResourcesServer) error`. The `AggregatedDiscoveryService_DeltaAggregatedResourcesServer` can send/recv DeltaDiscovery(Response|Request). Trying to figure out what [variant(s)](https://www.envoyproxy.io/docs/envoy/latest/api-docs/xds_protocol#four-variants) of xDS I'm looking at here. All 4?

Still pouring through the source just hoping for a quicker answer. Thanks